src/main/java/com/google/sps/app/servlets/GetTrip.java

```diff
@@ -0,0 +1,70 @@
+package com.google.sps.app.servlets;
+
+import com.google.cloud.datastore.BaseEntity;
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.Key;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.sps.model.Trip;
+import com.google.sps.util.dataStoreHelper;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/get-trip")
+public class GetTrip extends HttpServlet {
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String userID = request.getParameter("userID");
+        List<Trip> Trip = getTrip(userID);
+        Gson gson = new Gson();
+        response.setContentType("application/json;");
+        response.getWriter().println(gson.toJson(Trip));
+    }
+
+    public List<Trip> getTrip(String userID) {
+        QueryResults<?> results = queryDatastore(userID);
```

Comment:
There should be a try catch here.
It could look something like this: 
try {
    QueryResults<?> results = queryDatastore(userID);
} catch (IllegalArgumentException e) {
     return new ArrayList<Trip>(); //an empty list
}

then make sure that the doGet method checks for an empty string and returns a message like:
"There are no associated trips for this user. <a href=\"/NewTrip\">Please make a new Trip</a>"

Reply:
Thx, But I do think we don't need to redirect user to creating new trips, since In my "desgin", after user loged in. There is a page that display all the exisiting trips options for the user, and at the same time, offering create new trip. Thus, but only displaying "no trip exist" will do since this mesage will be displayed on the same page as creating new trip